lib/jekyll-admin/static_directory.rb

```diff
@@ -0,0 +1,89 @@
+module JekyllAdmin
+  class StaticDirectory
+
+    include Enumerable
+    include JekyllAdmin::URLable
+    include JekyllAdmin::APIable
+
+    DIRECTORY_TYPE = 'directory'
+    DIRECTORY_SEPARATOR = '/'
+
+    def initialize(path, config={})
+      @base = Pathname.new config.fetch(:base_directory)
+      @path           = Pathname.new path
+      @entries        = initialize_entries
+    end
+
+    def each
+      @entries.each do |entry|
+        yield entry
+      end
+    end
+
+    def to_liquid
+      {
+        basename: basename,
+        name: name,
+        extname: extname,
+        modified_time: modified_time,
+        path: relative_path,
+        type: DIRECTORY_TYPE,
+        entries: entries_to_array
+      }
+    end
+
+    def basename
+      @path.basename
+    end
+
+    def name
+      basename
+    end
+
+    def extname
+      nil
+    end
+
+    def modified_time
+      @path.mtime
+    end
+
+    def path
+      @path
+    end
+
+    def relative_path
+      @path.relative_path_from(@base).to_s
+    end
+
+    def resource_path
+      '/static_files/' + relative_path.to_s
+    end
+
+    def url
+      resource_path
+    end
```

Comment:
You'll want to differentiate between `http_url` and `api_url` as distinct methods. See https://github.com/jekyll/jekyll-admin/blob/master/lib/jekyll-admin/apiable.rb#L143-L149